kusto/ingest/internal/resources/resources.go

```diff
@@ -361,6 +366,11 @@ func (m *Manager) fetchRetry(ctx context.Context) {
 // Resources returns information about the ingestion resources. This will used cached information instead
 // of fetching from source.
 func (m *Manager) Resources() (Ingestion, error) {
+	lastFetchTime, ok := m.lastFetchTime.Load().(time.Time)
+	if !ok || lastFetchTime.Add(2*fetchInterval).After(time.Now().UTC()) {
+		m.fetchRetry(context.Background())
```

Comment:
Fetch retry may hang forever.
You should (at least optionally and here explicitly) limit the retry count (3?) and return an error if failed

Reply:
Huh, fetchRetry has an "attempts" variable that we increase, but never use..

Reply:
Fixed.